tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -912,6 +913,381 @@ void test_tls_pointer(void)
+	struct k_thread *thread;
+
+	thread = k_object_alloc(K_OBJ_THREAD);
+
+	/* failed due to run out off thread idx*/
+	zassert_not_null(thread, "alloc obj (0x%lx)\n", (uintptr_t)thread);
+	z_thread_perms_clear((void *)thread, curr_thread);
+	k_object_release((void *)thread);
+#endif
+
+	/* allocate kernel object */
+	p = k_object_alloc(K_OBJ_PIPE);
+	zassert_not_null(p, "alloc obj (0x%lx)\n", (uintptr_t)p);
+	s = k_object_alloc(K_OBJ_STACK);
+	zassert_not_null(s, "alloc obj (0x%lx)\n", (uintptr_t)s);
+	m = k_object_alloc(K_OBJ_MSGQ);
+	zassert_not_null(m, "alloc obj (0x%lx)\n", (uintptr_t)m);
+	q = k_object_alloc(K_OBJ_QUEUE);
+	zassert_not_null(q, "alloc obj (0x%lx)\n", (uintptr_t)q);
+
+	/* set permission opeation */
+	z_thread_perms_set((void *)p, curr_thread);
+	z_thread_perms_set((void *)s, curr_thread);
+	z_thread_perms_set((void *)q, curr_thread);
+	z_thread_perms_set((void *)m, curr_thread);
+
+	/* clear permission opeation */
+	z_thread_perms_clear((void *)p, curr_thread);
+	z_thread_perms_clear((void *)s, curr_thread);
+	z_thread_perms_clear((void *)q, curr_thread);
+	z_thread_perms_clear((void *)m, curr_thread);
+
+	/* release operations */
+	k_object_release((void *)p);
+	k_object_release((void *)s);
+	k_object_release((void *)q);
+	k_object_release((void *)m);
+
+	mslab = k_object_alloc(K_OBJ_MEM_SLAB);
+	zassert_not_null(mslab, "alloc obj (0x%lx)\n", (uintptr_t)mslab);
+	polls = k_object_alloc(K_OBJ_POLL_SIGNAL);
+	zassert_not_null(polls, "alloc obj (0x%lx)\n", (uintptr_t)polls);
+	timer = k_object_alloc(K_OBJ_TIMER);
+	zassert_not_null(timer, "alloc obj (0x%lx)\n", (uintptr_t)timer);
+
+	k_object_release((void *)mslab);
+	k_object_release((void *)polls);
+	k_object_release((void *)timer);
+
+	z_thread_perms_set((void *)mslab, curr_thread);
+	z_thread_perms_set((void *)polls, curr_thread);
+	z_thread_perms_set((void *)timer, curr_thread);
+
+	z_thread_perms_clear((void *)mslab, curr_thread);
+	z_thread_perms_clear((void *)polls, curr_thread);
+	z_thread_perms_clear((void *)timer, curr_thread);
+
+	/* no real object will be allocated */
+	ko = k_object_alloc(K_OBJ_ANY);
+	zassert_is_null(ko, "alloc obj (0x%lx)\n", (uintptr_t)ko);
+	ko = k_object_alloc(K_OBJ_LAST);
+	zassert_is_null(ko, "alloc obj (0x%lx)\n", (uintptr_t)ko);
+
+	/* alloc possible device driver */
+	ko = k_object_alloc(K_OBJ_LAST-1);
+	zassert_not_null(ko, "alloc obj (0x%lx)\n", (uintptr_t)ko);
+	z_thread_perms_set((void *)ko, curr_thread);
+	k_object_release((void *)ko);
+
+	/* error case of kobject free */
+	set_fault(K_ERR_CPU_EXCEPTION);
+	k_object_free(NULL);
+}
+
+/* @brief Test kernel object allocation and permission operations
+ *
+ * @details Create a dynamic kernel object with a unreasonable size,
+ * generate an expected out of memory error.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_dynamic_obj_create_out_of_memory(void)
+{
+	struct z_object *ko;
+
+	ko = z_dynamic_object_create(INT_MAX);
+	zassert_is_null(ko, "success is not reasonable");
```

Comment:
This message is confusing if you only look at serial output when this fails. Please state exactly what is wrong.

Reply:
Got it, indeed it is confusing. Thanks for correction this^^